gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py

```diff
@@ -0,0 +1,242 @@
+        info_keys_to_move_to_calls_regex='.*',
+        copy_quality_to_calls=True,
+        copy_filter_to_calls=True)
+    variants = self._get_sample_variants()
+
+    # Test single variant merge.
+    single_merged_variant = strategy.get_merged_variants([variants[0]])[0]
+    self.assertEqual(
+        [vcfio.VariantCall(name='Sample1', genotype=[0, 1],
+                           info={'GQ': 20, 'HQ': [10, 20],
+                                 'A1': 'some data', 'A2': ['data1', 'data2'],
+                                 ColumnKeyConstants.QUALITY: 2,
+                                 ColumnKeyConstants.FILTER: ['PASS']}),
+         vcfio.VariantCall(name='Sample2', genotype=[1, 0],
+                           info={'GQ': 10, 'FLAG1': True,
+                                 'A1': 'some data', 'A2': ['data1', 'data2'],
+                                 ColumnKeyConstants.QUALITY: 2,
+                                 ColumnKeyConstants.FILTER: ['PASS']})],
+        single_merged_variant.calls)
+
+    merged_variant = strategy.get_merged_variants(variants)[0]
+    self._assert_common_expected_merged_fields(merged_variant)
+    self.assertEqual(
+        [vcfio.VariantCall(name='Sample1', genotype=[0, 1],
+                           info={'GQ': 20, 'HQ': [10, 20],
+                                 'A1': 'some data', 'A2': ['data1', 'data2'],
+                                 ColumnKeyConstants.QUALITY: 2,
+                                 ColumnKeyConstants.FILTER: ['PASS']}),
+         vcfio.VariantCall(name='Sample2', genotype=[1, 0],
+                           info={'GQ': 10, 'FLAG1': True,
+                                 'A1': 'some data', 'A2': ['data1', 'data2'],
+                                 ColumnKeyConstants.QUALITY: 2,
+                                 ColumnKeyConstants.FILTER: ['PASS']}),
+         vcfio.VariantCall(name='Sample3', genotype=[1, 1],
+                           info={'A1': 'some data2', 'A3': ['data3', 'data4'],
+                                 ColumnKeyConstants.QUALITY: 20,
+                                 ColumnKeyConstants.FILTER: ['q10']}),
+         vcfio.VariantCall(name='Sample4', genotype=[1, 0],
+                           info={'GQ': 20,
+                                 'A1': 'some data2', 'A3': ['data3', 'data4'],
+                                 ColumnKeyConstants.QUALITY: 20,
+                                 ColumnKeyConstants.FILTER: ['q10']})],
+        merged_variant.calls)
+    self.assertEqual([], merged_variant.info.keys())
+
+  def test_get_snp_merge_keys(self):
+    strategy = merge_with_non_variants_strategy.MergeWithNonVariantsStrategy(
+        None, None, None, 2)
+
+    variant_1 = vcfio.Variant(reference_name='1', start=3, end=4)
+    variant_2 = vcfio.Variant(reference_name='2', start=4, end=5)
+
+    self.assertEqual(next(strategy.get_merge_keys(variant_1)), '1:2')
+    self.assertEqual(next(strategy.get_merge_keys(variant_2)), '2:4')
+
+  def test_get_nonv_merge_keys(self):
```

Comment:
nit: please use non_variant in the name.

Reply:
Done.